crates/sncast/src/starknet_commands/script.rs

```diff
@@ -34,21 +34,83 @@ use itertools::chain;
 use num_traits::ToPrimitive;
 use runtime::EnhancedHintError;
 use scarb_artifacts::ScarbCommand;
+use serde::Serialize;
 use sncast::helpers::response_structs::ScriptResponse;
 use sncast::helpers::scarb_utils::{
     get_package_metadata, get_scarb_manifest, get_scarb_metadata_with_deps, CastConfig,
 };
+use sncast::{print_formatted, stringify_command_result_struct, ValueFormat};
 use starknet::accounts::Account;
 use starknet::core::types::{BlockId, BlockTag::Pending, FieldElement};
 use starknet::providers::jsonrpc::HttpTransport;
 use starknet::providers::JsonRpcClient;
 use tokio::runtime::Runtime;
 
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Ord, PartialOrd)]
+pub enum Verbosity {
+    Quiet,
+
+    #[default]
+    Normal,
+}
+
+#[derive(Debug)]
+pub struct UI {
+    verbosity: Verbosity,
+    value_format: ValueFormat,
+    json: bool,
+}
+
+impl UI {
+    pub fn from_cli_args(is_quiet: bool, value_format: ValueFormat, json: bool) -> Self {
+        let verbosity = if is_quiet {
+            Verbosity::Quiet
+        } else {
+            Verbosity::Normal
+        };
+
+        Self {
+            verbosity,
+            value_format,
+            json,
+        }
+    }
+
+    pub fn is_quiet(&self) -> bool {
+        self.verbosity < Verbosity::Normal
+    }
+
+    pub fn print(&self, msg: &str) {
+        if self.verbosity >= Verbosity::Normal {
+            println!("{msg}");
+        }
+    }
+
+    pub fn print_cheatcode_response<T: Serialize>(
```

Comment:
We should have standarized structs for data we want to support printing.

Reply:
improved